metadata-ingestion/src/datahub/ingestion/source/dbt_manifest.py

```diff
@@ -0,0 +1,327 @@
+            dbtNode.upstream_urns = get_upstreams(
+                node['depends_on']['nodes'], 
+                nodes,
+                platform,
+                environment
+            )
+        else:
+            # It's a source 
+            dbtNode.materialization = catalog[key]['metadata']['type']
+            dbtNode.upstream_urns = []
+
+        if dbtNode.materialization != 'ephemeral':
+            dbtNode.columns = get_columns(catalog[dbtNode.dbt_name])
+        else:
+            dbtNode.columns = []
+
+        dbtNode.datahub_urn = get_urn_from_dbtNode(dbtNode.database, dbtNode.schema, dbtNode.relation_name, platform, environment)
+        
+        dbt_entities.append(dbtNode)
+
+    return dbt_entities
+
+
+def loadManifestAndCatalog(manifest_path, catalog_path, platform, environment) -> list[DBTNode]:
+    manifest_nodes = list[DBTNode]
+    
+    with open(manifest_path, "r") as manifest:
+        with open(catalog_path, "r") as catalog: 
+            dbt_manifest_json = json.load(manifest)
+            dbt_catalog_json = json.load(catalog)
+
+            manifest_nodes = dbt_manifest_json['nodes']
+            manifest_sources = dbt_manifest_json['sources']
+
+            all_manifest_entities = manifest_nodes | manifest_sources
+
+            catalog_nodes = dbt_catalog_json['nodes']
+            catalog_sources = dbt_catalog_json['sources']
+
+            all_catalog_entities = catalog_nodes | catalog_sources
+
+            logger.info('loading {} nodes, {} sources'.format(len(all_manifest_entities), len(all_catalog_entities)))
+    
+            nodes = extract_dbt_entities(all_manifest_entities, all_catalog_entities, platform, environment)
+
+            return nodes
+
+def get_urn_from_dbtNode(database: str, schema: str, relation_name: str, platform: str, env:str) -> str:
+    db_fqn = f"{database}.{schema}.{relation_name}".replace('"', '')
+    return f"urn:li:dataset:(urn:li:dataPlatform:{platform},{db_fqn},{env})"
+
+def get_custom_properties(node: DBTNode) -> List[dict[str,str]]:
+    tags = {}
+    tags['dbt_node_type'] = node.node_type
+    tags['materialization'] = node.materialization
+    tags['dbt_file_path'] = node.dbt_file_path
+    return tags
+
+def get_upstreams(upstreams: List[str], all_nodes, platform: str, environment: str) -> List[str]:
+    upstream_urns = []
+
+    for upstream in upstreams:
+        upstream_node = all_nodes[upstream]
+
+        upstream_urns.append(get_urn_from_dbtNode(
+            upstream_node['database'],
+            upstream_node['schema'],
+            upstream_node['relation_name'],
+            platform,
+            environment
+        ))
+
+    return upstream_urns
+
+def get_upstream_lineage(upstream_urns: List[str]) -> UpstreamLineage:
+    ucl: List[uc] = []
+
+    for dep in upstream_urns:
+        uc = UpstreamClass(
+        dataset=dep, 
+        auditStamp=AuditStamp(
+            actor="urn:li:corpUser:dbt_executor",
```

Comment:
Not sure what the default ought to be here, I'm leaning towards a configurable parameter in the config file.

Reply:
Yep that should work. The SQL sources currently have this hardcoded, but I'm planning on making it configurable there too